Hi, different solution I found out to be working. Please take a look.

Stefan-23 opened the pull request:
Hi, different solution I found out to be working. Please take a look. I couldn't make first solution to work but this works as i should.

Stefan-23 committed:
Hi, different solution I found out to be working. Please take a look.
I couldn't make first solution to work but this works as i should.

Marko101ish commented on include/PlayerActions.h:
```diff
@@ -0,0 +1,37 @@
+
+enum class side { LEFT, RIGHT, NONE };
+
+void handleChopRight(
```
I have a few questions here, why do you pass all parameters by reference & ?

EDIT:
Okay, I see, all of these are meant to be set in the method itself

Marko101ish commented on include/functions.h:
```diff
@@ -0,0 +1,50 @@
+
+
+
+void handleInput(side playerDirection)
```
So we still need this one or no?

Marko101ish commented on Timber.cpp:
```diff
@@ -313,71 +307,40 @@ int main()
-                    break;
-                }
-                // chop.play();
+                handleChopRight(
```
As discussed in person, this doesn't address the core issue. Code is still copy-pasted but just moved into two separate functions instead of main.

Since classes and objects are off the table for now, it would be better to declare the variables you need in global space so your functions can reach them instead of owning them in main and pass them by ref (something along the lines of handleInput() you implemented that is unused right now)

Now since you mentioned you had troubles with handleInput(), here's some more info on using the debugger which should come in handy for solving it:

https://m.youtube.com/watch?v=0ebzPwixrJA

Main part is in the first half labeled Breakpoints, the second half is about memory, it's a bit more advanced, you likely won't need it right now, but it is extremely useful, as well.

In general all videos in this playlist are fair game up until classes, so feel free to give those a watch as well, if you have time:

https://m.youtube.com/playlist?list=PLlrATfBNZ98dudnM48yfGUldqGD0S4FFb

And as always ask if you get confused or need help

Stefan-23 committed:
Input Function implemented

Hello, without using Mika solution I managed to implement my first better function as we talked
